Add bitmap and raw bitmap byte phash support.

ThatRubenAguilar opened the pull request:
I had a need to phash bitmaps and subsections of the same bitmap, so I added support for those cases. I added comparison testing of these different methods to TestApp to assure they at least match the original bitmap source implementation.

If there's anything you want to discuss/change let me know.

ThatRubenAguilar committed:
[bitmaps] Add bitmap support, update dependencies in projects

ThatRubenAguilar committed:
[bitmaps] Update TestApp to verify identical results from run, account for stride bytes in bitmap computation, add bitmap helper methods

ThatRubenAguilar committed:
[bitmaps] Fix bitmap pixelwidth not being converted to bytewidth on calculating stride delta, add rawbyte and rectangle area manipulation for faster consecutive calls on subsections of the same image

ThatRubenAguilar committed:
[bitmaps] Add raw bitmap data phash calculations, clean up naming

ThatRubenAguilar committed:
[bitmaps] Add bitmap raw comparisons to test app

ThatRubenAguilar committed:
[bitmaps] Correct Byte Per Pixel calculations

pgrho commented:
Thank you for sending this request. But I am using this library in an ASP.NET Web application. And the dependency for System.Drawing.dll the added in this Pull Request are not supported to use in IIS. 

> [System.Drawing Namespace](https://msdn.microsoft.com/en-us/library/system.drawing(v=vs.110).aspx)
> Classes within the System.Drawing namespace are not supported for use within a Windows or ASP.NET service. Attempting to use these classes from within one of these application types may produce unexpected problems, such as diminished service performance and run-time exceptions. For a supported alternative, see Windows Imaging Components.

Can you rewrite those dependencies to use `System.Windows.Media.Imaging` of PresentationCore.dll that is based on the Windows Imaging Component? Or you may move GDI+ calls to new System.Drawing dependent project.

pgrho commented on src/Shipwreck.Phash/ImagePhash.cs:
```diff
@@ -73,11 +74,56 @@ public static Digest ComputeDigest(string fileName, double sigma = DEFAULT_SIGMA
+            return ComputeDigest(bf, sigma, gamma, numberOfAngles: numberOfAngles);
+        }
+
+        public static Digest ComputeDigest(BitmapSource bitmapSource, double sigma = DEFAULT_SIGMA, double gamma = DEFAULT_GAMMA, int numberOfAngles = DEFAULT_NUMBER_OF_ANGLES)
```
Change method name like `ComputeBitmapDigest` to avoid PresentationCore dependent signature to be included in method group. Such overload forces non-WPF user to add references explicitly.

ThatRubenAguilar commented:
I had a feeling about that when I saw the TestApp dependencies change. I'll move the dependent calls to a project Shipwreck.Phash.Bitmaps and they can live as two packages.

ThatRubenAguilar committed:
[bitmaps] Move bitmap dependencies to Shipwreck.Phash.Bitmaps project

ThatRubenAguilar committed:
[bitmaps] Convert Bitmaps to SDK project

ThatRubenAguilar committed:
[bitmaps] Output xml documentation file, suppress public comment warning

ThatRubenAguilar committed:
[bitmaps] Add pack target

ThatRubenAguilar commented:
Moved bitmap dependencies to Shipwreck.Phash.Bitmaps and rebased on master. Added some pack targets for msbuild via nuget as from what I've read nuget 4.0+ doesnt work with the new csproj format for now.

pgrho commented:
Published packages.
